tests/unit/jobs/test_views.py

```diff
@@ -841,3 +862,47 @@ def test_get_submission_by_pk_when_user_is_neither_challenge_host_nor_submission
         response = self.client.get(self.url)
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class ChallengeLeaderboardTest(BaseAPITestClass):
+
+    def setUp(self):
+        super(ChallengeLeaderboardTest, self).setUp()
+
+        self.submission = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.user1,
+            status="finished",
+            input_file=self.challenge_phase.test_annotation,
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+        )
+
+        self.submission.is_public = True
+        self.submission.status = Submission.FINISHED
```

Comment:
I think you set the submission status to finished in line 876. Why are you setting it again? Please let me know if I am missing some context here.

Reply:
When a submission is created for first time in `jobs/models.py` save method is written such that it sets status as `SUBMITTED` only.

Reply:
Which is why I have to explicitly set the status